src/test/disconnected_transactions.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+//
+#include <test/util/setup_common.h>
+#include <kernel/disconnected_transactions.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(disconnected_transactions, TestChain100Setup)
+
+//! Tests that DisconnectedBlockTransactions limits its own memory properly
+BOOST_AUTO_TEST_CASE(disconnectpool_memory_limits)
+{
+    // Use the coinbase transactions from TestChain100Setup. It doesn't matter whether these
+    // transactions would realistically be in a block together, they just need distinct txids for
+    // this test to work.
+    std::vector<CTransactionRef> block_vtx(m_coinbase_txns);
+
+    size_t usage_full{0};
+    // DisconnectedBlockTransactions with a comfortable maximum memory usage s.t. nothing is evicted.
```

Comment:
nit: less abbreviations if not necessary
```suggestion
    // DisconnectedBlockTransactions with a comfortable maximum memory usage so that nothing is evicted.
```